pkg/sql/compile/compile.go

```diff
@@ -1857,7 +1859,7 @@ func (c *Compile) generateNodes(n *plan.Node) (engine.Nodes, error) {
 	var err error
 	var db engine.Database
 	var rel engine.Relation
-	var ranges [][]byte
+	var ranges any
```

Comment:
why do you need any?    [][]byte is bad, any is just worse.    Can we get a real type

Reply:
to support both disttae and memory engine